packages/fxa-settings/src/lib/gql.ts

```diff
@@ -1,8 +1,32 @@
 import { ApolloClient, createHttpLink, from } from '@apollo/client';
 import { setContext } from '@apollo/client/link/context';
-import { onError } from '@apollo/client/link/error';
+import { ErrorHandler, onError } from '@apollo/client/link/error';
 import { cache, sessionToken, typeDefs } from './cache';
 
+export const errorHandler: ErrorHandler = ({ graphQLErrors, networkError }) => {
+  let reauth = false;
+  const hasFlowParams = /flowBeginTime/.test(window.location.search);
+  if (graphQLErrors) {
+    for (const error of graphQLErrors) {
+      if (error.extensions?.code === 'UNAUTHENTICATED') {
+        reauth = true;
+      }
+    }
+  }
+  if (networkError && 'statusCode' in networkError) {
+    if (networkError.statusCode === 401) {
+      reauth = true;
+    }
+  }
+  if (reauth && !hasFlowParams) {
```

Comment:
this `hasFlowParams` bit is the part that changed

Reply:
I'm not sure `hasFlowParams` is the right thing to check, or if any addition here is necessary... iiuc the infinite loop was caused by the gql-api server not being able to access the auth-server, and it only sending auth errors. By returning an ApolloError for non-auth errors I think we've fixed that problem. I can imagine a situation where we get a legit auth error and do have flowParams where we still want to reauth instead of showing an error... like a page refresh after you deleted that session elsewhere

tldr: i think the gql server change is all we need/want

Reply:
Good point. We definitely might need to reauth when there are flow params. And I'm not sure the flow params should be in the query string anyway, which means this would need to be tweaked whenever that other change happens. I'll remove this bit from the PR.

Reply:
I'm still getting some weird reload loop if I nav to localhost:3000/beta/settings locally. Is that just me?

Reply:
lol never mind, I should be going to localhost:**3030**/beta/settings 🤦